apps/govern/components/Contracts/EditVotes.tsx

```diff
@@ -0,0 +1,409 @@
+        chainId: newVote.chainId,
+        weight: `${Math.floor(newVote.weight * 100)}`,
+      });
+    } else {
+      // Otherwise we need to remove weight from that contract
+      // before voting on the others contacts
+      const chainId = stakingContracts.find((item) => item.address === oldVoteAddress)?.chainId;
+      if (chainId) {
+        newVotes.unshift({ address: oldVoteAddress as Address, chainId, weight: '0' });
+      }
+    }
+  });
+
+  // Add remaining new votes to the result
+  sortedAllocations.forEach((newVote) => {
+    if (newVotes.findIndex((item) => item.address === newVote.address) === -1) {
+      newVotes.push({
+        address: newVote.address,
+        chainId: newVote.chainId,
+        weight: `${Math.floor(newVote.weight * 100)}`,
+      });
+    }
+  });
+
+  return newVotes;
+};
+
+const TotalPower = styled(Text)`
+  font-size: 24px;
+  font-weight: 700;
+  line-height: 32px;
+  margin-bottom: 16px;
+`;
+
+type EditVotesProps = {
+  allocations: Allocation[];
+  setAllocations: Dispatch<SetStateAction<Allocation[]>>;
+  setIsUpdating: Dispatch<SetStateAction<boolean>>;
+};
+
+const getColumns = (
+  allocations: Allocation[],
+  setAllocation: (value: number, index: number) => void,
+  removeAllocation: (index: number) => void,
+  isError: boolean,
+): ColumnsType<Allocation> => [
+  {
+    title: 'Contract name',
+    key: 'name',
+    render: (_, record) => <Text strong>{record.metadata.name}</Text>,
+    width: 200,
+  },
+  {
+    title: 'Chain',
+    key: 'chain',
+    render: (_, record) => (
+      <Text type="secondary">{CHAIN_NAMES[record.chainId] || record.chainId}</Text>
+    ),
+    width: 120,
+  },
+  {
+    title: 'My voting weight',
+    key: 'weight',
+    render: (_, record, index) => (
+      <Flex gap={16}>
+        <InputNumber
+          addonAfter="%"
+          min={0}
+          max={100}
+          step={0.01}
+          value={allocations[index].weight}
+          status={isError ? 'error' : undefined}
+          onChange={(value) => {
+            if (value) {
+              setAllocation(value, index);
+            }
+          }}
+        />
+        <Button
+          icon={<DeleteOutlined />}
+          onClick={() => removeAllocation(index)}
+          style={{ flex: 'none' }}
+        />
+      </Flex>
+    ),
+    width: 140,
+  },
+];
+
+const ConfirmModal = ({
+  isOpen,
+  handleOk,
+  handleClose,
+  isLoading,
+  allocationsLength,
+  totalPower,
+}: {
+  isOpen: boolean;
+  handleOk: () => void;
+  handleClose: () => void;
+  isLoading: boolean;
+  allocationsLength: number;
+  totalPower: number;
+}) => {
+  return (
+    <Modal
+      title="Confirm voting weight update"
+      open={isOpen}
+      onOk={handleOk}
+      onCancel={handleClose}
+      cancelText="Cancel"
+      okText="Confirm voting weight"
+      confirmLoading={isLoading}
+    >
+      <Paragraph>
+        {`You're allocating ${parseFloat(
+          (totalPower / 100).toFixed(2),
```

Comment:
without full context, this may not produce the desired result

perhaps we could rename `totalPower` to `allocatedPower`, and do something like `allocatedPower/totalPower`?

may not apply though, let me know